packages/chai/package.py

```diff
@@ -120,13 +126,13 @@ class Chai(CachedCMakePackage, CudaPackage, ROCmPackage):
         description="Tests to run",
     )
 
-    depends_on("cmake@3.8:", type="build")
-    depends_on("cmake@3.9:", type="build", when="+cuda")
-    depends_on("cmake@3.14:", type="build", when="@2022.03.0:")
+    depends_on("cmake@3.23:", type="build", when="@2024.07.0:")
+    depends_on("cmake@3.14", type="build", when="@2022.03:2024.02")
```

Comment:
@rhornung67 it does not seem reasonable to me to pin the CMake version like so. It would force a CMake rebuild when it is probably not necessary.
Please let me know if you think otherwise.

Reply:
Whatever you think is best works for me. I was attempting to follow the changes to the cmake minimum required version in the top-level CHAI CMakeLists.txt file from one release to the next.  Similarly for other projects.

Reply:
OK, my suggestion consists in a minute change: `cmake@3.14` -> `cmake@3.14:`, I’ll apply it during the merge.

Reply:
Thank you @adrienbernede. Would it be a good idea to document best practices someplace for things like this related to maintaining a project spack package?

Reply:
Of course it would, of course... Let me finish that big RADIUSS packages sync effort I am deep into at the moment and then I’ll settle down and think.

Reply:
Maybe we could work on it together with @white238 since he's the WSC DevOps lead? It would be good to have all of our projects follow similar spack package maintenance practices.